### ISSUE TITLE
HQ: a warning about undefined nodata is too alarming

### ISSUE THREAD

**Issue by davemfish** [natcap/invest#1039]
As discussed here: https://community.naturalcapitalproject.org/t/warning-raster-has-undefined-nodata-value-in-invest-3-11-0-workbench/2937

Here's the warning and it's context:

```python
def _raster_values_in_bounds(raster_path_band, lower_bound, upper_bound):
    """Check raster values are between ``lower_bound`` and ``upper_bound``.

    Check that the raster has values ``lower_bound`` <= x <= ``upper_bound``.
    Nodata values are ignored.

    Args:
        raster_path_band (tuple): a 2 tuple for a GDAL raster path with
            the form (filepath, band index) to the raster on disk.
        lower_bound (int): integer for the lower bound of raster values,
            inclusive.
        upper_bound (int): integer for the upper bound of raster values,
            inclusive.

    Returns:
        True if values are within range and False otherwise.
    """
    raster_info = pygeoprocessing.get_raster_info(raster_path_band[0])
    raster_nodata = raster_info['nodata'][0]

    if raster_nodata is None:
        LOGGER.warning(
            f"Raster has undefined NODATA value for {raster_path_band[0]}.")
        # If raster nodata is None then set to _OUT_NODATA to use for masking
        # where in this case nodata_mask will be all False.
        raster_nodata = _OUT_NODATA

    values_valid = True

    for _, raster_block in pygeoprocessing.iterblocks(raster_path_band):
        nodata_mask = ~utils.array_equals_nodata(raster_block, raster_nodata)
        if ((raster_block[nodata_mask] < lower_bound) |
                (raster_block[nodata_mask] > upper_bound)).any():
            values_valid = False
            break

    return values_valid
```

I was going to suggest just changing it to `debug`, but actually I think this function might work even better if `raster_nodata` is left as `None` so long as `utils.array_equals_nodata` can handle `None`. And it just so happens that James is making this possible in https://github.com/natcap/invest/pull/1026

So now I think my suggestion is to wait for that PR, and then remove this whole chunk:
```python
    if raster_nodata is None:
        LOGGER.warning(
            f"Raster has undefined NODATA value for {raster_path_band[0]}.")
        # If raster nodata is None then set to _OUT_NODATA to use for masking
        # where in this case nodata_mask will be all False.
        raster_nodata = _OUT_NODATA
```

As it is, there is the _extremely unlikely_ case where a value of `_OUT_NODATA` was actually present in the raster, but will now get masked out allowing this function to return `True` when it should not.

What do you think @dcdenu4 ?

**Comment by dcdenu4** [comment]
That sounds right to me! Thanks Dave.

**Comment by davemfish** [comment]
We ended up merging the aforementioned `utils.array_equals_nodata` change into `release/3.12` branch. So we could at any time address this issue on that branch. We expect the next release to be `3.12` so there's no advantage to addressing this on `main`.

**Comment by phargogh** [comment]
Resolved in https://github.com/natcap/invest/pull/1056 and released in InVEST 3.12.
